docs/examples/plot_dynamic_beat.py

```diff
@@ -0,0 +1,132 @@
+# --------------------------
+# If we run the beat tracker on this signal directly and sonify the result as a click
+# track, we can see that it does not follow particularly well.
+# This is because the beat tracker is assuming a single (average) tempo across the
+# entire signal.
+# Note: here we are disabling trimming because we know the beats carry through to
+# the end of the signal.
+
+tempo, beats_static = librosa.beat.beat_track(y=y, sr=sr, units='time', trim=False)
+
+click_track = librosa.clicks(times=beats_static, sr=sr, click_freq=660,
+                             click_duration=0.25, length=len(y))
+
+print(f"Tempo estimate: {tempo[0]:.2f} BPM")
+Audio(data=y+click_track, rate=sr)
+
+#########################################
+# -------------
+# Dynamic tempo
+# -------------
+# Instead of using a single tempo, we can supply an array of tempo estimates
+# to the beat tracker.  This can be computed by disabling aggregation in the
+# tempo estimator as follows.
+
+tempo_dynamic = librosa.feature.tempo(y=y, sr=sr, aggregate=None, std_bpm=4)
+
+# %%
+# Here, we've expanded the standard deviation of the tempo estimator from its
+# default of 1 to 4.  This is to account for the broad range of tempo drift in this
+# particular recording (30 BPM to 240 BPM).
+# We can plot the dynamic tempo estimate, along with the static estimate, to see how
+# they differ.
+
+fig, ax = plt.subplots()
+times = librosa.times_like(tempo_dynamic, sr=sr)
+ax.plot(times, tempo_dynamic, label='Dynamic tempo estimate')
+ax.axhline(tempo, label='Static tempo estimate', color='r')
+ax.legend()
+ax.set(xlabel='Time [sec]', ylabel='Tempo [BPM]')
+
+# %%
+# The dynamic tempo estimate is computed (by default) using a 8-second sliding
+# window.  It is clearly not perfect, containing many jagged and non-uniform steps,
+# but it does provide a rough picture of how tempo is changing over time.
+#
+# We can now use this dynamic tempo in the beat tracker directly:
+
+tempo, beats_dynamic = librosa.beat.beat_track(y=y, sr=sr, units='time',
+                                               bpm=tempo_dynamic, trim=False)
+
+click_dynamic = librosa.clicks(times=beats_dynamic, sr=sr, click_freq=660,
+                               click_duration=0.25, length=len(y))
+
+Audio(data=y+click_dynamic, rate=sr)
+
+# %%
+# (Note that since we're providing the tempo estimates as input to the beat tracker,
+# we can ignore the first return value (`tempo`) which will simply be a copy of the
+# input (`tempo_dynamic`).)
+#
+# We can visualize the difference in estimated beat timings as follows:
+
+fig, ax = plt.subplots()
+librosa.display.waveshow(y=y, sr=sr, ax=ax, color='k', alpha=0.7)
+ax.vlines(beats_static, 0.75, 1., label='Static tempo beat estimates', color='r')
+ax.vlines(beats_dynamic, -1, -0.75, label='Dynamic tempo beat estimates', color='b', linestyle='--')
+ax.legend()
+
+
+# %%
+# In the figure above (and sonified click track), it can be observed that the
```

Comment:
In the figure above, we observe that `beats_dynamic`, the beat estimates derived from dynamic tempo, align well to the audio recording. On the contrary, `beats_static` suffers from alignment problems. Indeed, it has too many detections towards the beginning of the recording and too few detections towards the end of the recording.